extra_data/tests/test_reader_mockdata.py

```diff
@@ -615,13 +615,24 @@ def test_open_file(mock_sa3_control_data):
 
 
 def test_permission():
+    #  If users is root then `PermissionError` won't be raised, set the euid to
+    #  standard user so the tests work correctly
+    euid_was_root = False
+    if hasattr(os, 'seteuid'):
+        if os.geteuid() == 0:
+            os.seteuid(1000)
+            euid_was_root = True
+
     d = mkdtemp()
     os.chmod(d, not stat.S_IRUSR)
     with pytest.raises(PermissionError) as excinfo:
         run = RunDirectory(d)
     assert "Permission denied" in str(excinfo.value)
     assert d in str(excinfo.value)
 
+    if euid_was_root:
+        os.seteuid(0)
```

Comment:
This should be in a `finally:` block, so it gets reset even if the test fails. Or make it a pytest fixture.

But to be honest, it may be simpler to just skip the test if run as root. It's getting tested regularly as non-root, and it's not a particularly critical test.

Reply:
Made it so that the test is skipped just skipped if euid is 0 now